### ISSUE TITLE
Inlining failed in call to 'always_inline' when building the project

### ISSUE THREAD

**Issue by alexmikhalevich** [risc0/risc0#50]
I try to build and run all tests by executing `bazelisk test //...` but then I get a compilation error:

```
ERROR: /home/kamos/.cache/bazel/_bazel_kamos/76ead71fc5575ebd7e484e2e8c787a36/external/.workspace.oneTBB/BUILD.bazel:25:11: Compiling src/tbb/task_dispatcher.cpp [for host] failed: (Exit 1): gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 31 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/x86gprintrin.h:81,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/immintrin.h:27,
                 from external/.workspace.oneTBB/include/oneapi/tbb/detail/_machine.h:42,
                 from external/.workspace.oneTBB/include/oneapi/tbb/detail/_utils.h:26,
                 from external/.workspace.oneTBB/src/tbb/task_dispatcher.h:20,
                 from external/.workspace.oneTBB/src/tbb/task_dispatcher.cpp:17:
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/waitpkgintrin.h: In function 'void tbb::detail::r1::prolonged_pause()':
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/waitpkgintrin.h:53:1: error: inlining failed in call to 'always_inline' 'unsigned char _tpause(unsigned int, long long unsigned int)': target specific option mismatch
   53 | _tpause (unsigned int __A, unsigned long long __B)
      | ^~~~~~~
In file included from external/.workspace.oneTBB/src/tbb/task_dispatcher.h:24,
                 from external/.workspace.oneTBB/src/tbb/task_dispatcher.cpp:17:
external/.workspace.oneTBB/src/tbb/scheduler_common.h:230:16: note: called from here
  230 |         _tpause(0, time_stamp + 700);
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/x86gprintrin.h:81,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/immintrin.h:27,
                 from external/.workspace.oneTBB/include/oneapi/tbb/detail/_machine.h:42,
                 from external/.workspace.oneTBB/include/oneapi/tbb/detail/_utils.h:26,
                 from external/.workspace.oneTBB/src/tbb/task_dispatcher.h:20,
                 from external/.workspace.oneTBB/src/tbb/task_dispatcher.cpp:17:
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/waitpkgintrin.h:53:1: error: inlining failed in call to 'always_inline' 'unsigned char _tpause(unsigned int, long long unsigned int)': target specific option mismatch
   53 | _tpause (unsigned int __A, unsigned long long __B)
      | ^~~~~~~
In file included from external/.workspace.oneTBB/src/tbb/task_dispatcher.h:24,
                 from external/.workspace.oneTBB/src/tbb/task_dispatcher.cpp:17:
external/.workspace.oneTBB/src/tbb/scheduler_common.h:230:16: note: called from here
  230 |         _tpause(0, time_stamp + 700);
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/x86gprintrin.h:81,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/immintrin.h:27,
                 from external/.workspace.oneTBB/include/oneapi/tbb/detail/_machine.h:42,
                 from external/.workspace.oneTBB/include/oneapi/tbb/detail/_utils.h:26,
                 from external/.workspace.oneTBB/src/tbb/task_dispatcher.h:20,
                 from external/.workspace.oneTBB/src/tbb/task_dispatcher.cpp:17:
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/waitpkgintrin.h:53:1: error: inlining failed in call to 'always_inline' 'unsigned char _tpause(unsigned int, long long unsigned int)': target specific option mismatch
   53 | _tpause (unsigned int __A, unsigned long long __B)
      | ^~~~~~~
In file included from external/.workspace.oneTBB/src/tbb/task_dispatcher.h:24,
                 from external/.workspace.oneTBB/src/tbb/task_dispatcher.cpp:17:
external/.workspace.oneTBB/src/tbb/scheduler_common.h:230:16: note: called from here
  230 |         _tpause(0, time_stamp + 700);
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~
INFO: Elapsed time: 45.283s, Critical Path: 28.89s
INFO: 99 processes: 29 internal, 70 linux-sandbox.
FAILED: Build did NOT complete successfully
```

Is this somewhat known and already solved?

The system is archlinux, kernel 5.17.1, gcc version 11.2.0.

**Comment by jbruestle** [comment]
So this seems to be an issue with building tbb, which in an Intel threading library we rely on for CPU accelerations (https://www.intel.com/content/www/us/en/developer/tools/oneapi/onetbb.html#gs.x331ei).  Looks like it's related to the compiler version or flags, perhaps the lack of the _tpause intrinsic due to the march (https://www.felixcloutier.com/x86/tpause)?  But we don't currently have a repo on any of our test machines, and given that it's in a third party dependency it might take a bit to track down.  I don't think we should be setting march=native anywhere, but just in case that's sneaking in somewhere, what CPU are you building on?  I suspect if we grab gcc 11.2.0 our chances of getting it to reproduce locally are decent.

**Comment by alexmikhalevich** [comment]
I am building this on 12th Generation Intel Core i7-12700H.

**Comment by flaub** [comment]
This seems to be related: https://github.com/oneapi-src/oneTBB/issues/370

**Comment by flaub** [comment]
@alexmikhalevich Could you try this branch? I don't have an archlinux machine handy to test this: flaub/fix-50

**Comment by flaub** [comment]
@alexmikhalevich Could you try again with this branch: `flaub/fix-50`?

**Comment by alexmikhalevich** [comment]
now works, thanks

**Comment by flaub** [comment]
Fixed with #53
